### PR TITLE
Shira abrams i18n

### PR DESCRIPTION
**Pull request by Shira-Abrams**
i add i18next library 
add resurce file 
and i  did small example  on one page

### CODE REVIEW

File: client-side/src/components/settings/Settings.jsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next'
+import GenericButton  from '../../stories/Button/GenericButton.jsx';
+import GenericInput from '../../stories/GenericInput/genericInput.jsx'
+import Select from '../../stories/Select/Select.jsx'
+import {CHANGE_RINGTONE,SEND_PREFERENCE,SELECT_LANGUAGES} from './constantSetting.js'
+
+const Setting = ({currentUser}) => {
```
Comment (moriya2004) [inline client-side/src/components/settings/Settings.jsx:10]: change the Setting to Settings

File: client-side/src/App.jsx [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 function App() {
+
+
+  
   return (
     <div className="App">
       <Header/>
-      <RingtoneEditButton/>
-      <ManageNotifications></ManageNotifications>
+       <Setting currentUser={{preferences:{_id:'fdfsd',emailFrequency:'never'}}}/>
```
Comment (moriya2004) [inline client-side/src/App.jsx:14]: delete it

File: client-side/src/components/settings/Settings.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const {emailFrequency,sendNotificationTime,_id}= currentUser.preferences
```
Comment (moriya2004) [inline client-side/src/components/settings/Settings.jsx:11]: change to .preference

File: client-side/src/components/settings/Settings.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    const url=process.env.REACT_APP_BASE_URL;
+    const [ringtoneFile, setRingtoneFile] = useState(null);
+    const [audioSrc,setAudioSrc]  = useState();
+    const [lng,setLng] = useState('en');
```
Comment (moriya2004) [inline client-side/src/components/settings/Settings.jsx:15]: Choose a meaningful name for the variable

File: client-side/src/components/settings/Settings.jsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          </div>
+          <div>
+          { audioSrc &&
+            <audio controls>
+               <source src={audioSrc} ></source>
+            </audio>}
+          </div>       
+
+         <Select  title={t(SELECT_LANGUAGES)} 
+          options={[{text:'עברית',value:'he'}, {text:'española',value:'es'},{text:'english',value:'en',}]}
```
Comment (moriya2004) [inline client-side/src/components/settings/Settings.jsx:68]: use enum

File: client-side/src/components/settings/Settings.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+          <div>
+          { audioSrc &&
+            <audio controls>
+               <source src={audioSrc} ></source>
+            </audio>}
+          </div>       
+
+         <Select  title={t(SELECT_LANGUAGES)} 
+          options={[{text:'עברית',value:'he'}, {text:'española',value:'es'},{text:'english',value:'en',}]} 
+          className=''
```
Comment (moriya2004) [inline client-side/src/components/settings/Settings.jsx:69]: Choose a meaningful name for the className

File: client-side/src/components/settings/Settings.jsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+          widthOfSelect='200px'
+          value={lng}
+          onChange={e=>handleLngChange(e.target.value)}/>
+        </div>
+        
+        
+        
+  
+          <GenericButton size='small'  label={t(SEND_PREFERENCE)} onClick={sendPreference} className='' disabled={false}/>
+      </>
```
Comment (moriya2004) [inline client-side/src/components/settings/Settings.jsx:80]: Choose a meaningful name for the className

File: client-side/src/components/settings/constantSetting.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export const SEND_PREFERENCE='send preferences';
+export const CHANGE_RINGTONE='change-ringtone';
+export const SELECT_LANGUAGES='sleect-language'
```
Comment (moriya2004) [inline client-side/src/components/settings/constantSetting.js:3]: use one export for them all

**Review by moriya2004 — COMMENTED** [review]
fix your project

**Review by moriya2004 — APPROVED** [review]

File: client-side/src/App.jsx [hunk 9]
```diff
@@ -1,19 +1,21 @@
 import React from 'react';
 import { Provider } from 'react-redux';
 import { RouterProvider } from 'react-router-dom';
-import Footer from './stories/footer/FooterComponent';
 import { router } from './router/router.jsx';
 import { store } from './redux/store.jsx';
+import Footer from './stories/footer/FooterComponent';
 import { SnackbarProvider } from 'notistack';
```
Comment (e171271) [inline client-side/src/App.jsx:7]: re-order the imports

File: client-side/src/App.jsx [hunk 10]
```diff
@@ -1,19 +1,21 @@
...
 import './App.scss';
-
 function App() {
+  
+  
   return (
 
-    <>
+
```
Comment (e171271) [inline client-side/src/App.jsx:14]: remove the empty line breaks

File: client-side/src/stories/Button/GenericButton.scss [hunk 11]
```diff
@@ -19,8 +19,8 @@
             background-color:  white;
             color: black ;
         } 
+        
     }
-    
    
-
+
```
Comment (e171271) [inline client-side/src/stories/Button/GenericButton.scss:25]: remove empty line breaks

File: client-side/src/stories/GenericInput/genericInput.scss [hunk 12]
```diff
@@ -1,3 +1,4 @@
+/* genericInput.scss */
```
Comment (e171271) [inline client-side/src/stories/GenericInput/genericInput.scss:1]: remove comments

File: client-side/src/stories/GenericInput/genericInput.scss [hunk 13]
```diff
@@ -69,4 +70,46 @@
...
+.file-upload{
+  display: inline-block;
+  .generic-input-file{
+    border-radius: 8px;
+    transition: border-color 0.3s ease, box-shadow 0.3s ease;
+    border-color: $color-border;
+    border-width: 1px;
+    background-color:$color-border ;
+    color: $color-black;
+    font-family: 'Montserrat', sans-serif;
```
Comment (e171271) [inline client-side/src/stories/GenericInput/genericInput.scss:85]: font-family should come from variables file

File: client-side/src/stories/header/header.jsx [hunk 14]
```diff
@@ -1,16 +1,35 @@
...
 import MessageIcon from './Icon'
 import './header.scss';
 import { selectAuth } from '../../redux/auth/auth.selector';
+
+import './header.scss';
+
 function ResponsiveAppBar() {
   const [anchorElNav, setAnchorElNav] = useState(null);
   const [anchorElUser, setAnchorElUser] = useState(null);
+  const { t, i18n } = useTranslation();
```
Comment (e171271) [inline client-side/src/stories/header/header.jsx:29]: use meaningful names

**Review by e171271 — CHANGES_REQUESTED** [review]

File: client-side/src/stories/header/header.jsx (hunk 14 above)
Comment (Shira-Abrams) [reply client-side/src/stories/header/header.jsx:29]: its a library function  its constant name should i change it ?

Comment (e171271) [reply client-side/src/stories/header/header.jsx]: you can change the name when you sdstructuring it:
const {t: translate}=useTranslation();

File: client-side/src/stories/header/header.jsx [hunk 15]
```diff
@@ -1,16 +1,32 @@
...
 import './header.scss';
 import { selectAuth } from '../../redux/auth/auth.selector';
+
+import './header.scss';
+
 function ResponsiveAppBar() {
   const [anchorElNav, setAnchorElNav] = useState(null);
   const [anchorElUser, setAnchorElUser] = useState(null);
-  const { user } = useSelector(selectAuth); 
+  const { t: translate } = useTranslation();  const { user } = useSelector(selectAuth);
```
Comment (SariGlick) [inline client-side/src/stories/header/header.jsx:27]: add line break between the consts

File: server-side/.env [hunk 16]
```diff
@@ -1,5 +1,5 @@
-PORT=3000
+PORT=5000
 password=EuACQK5n92JI3vn4
-DB_URL="mongodb://127.0.0.1:27017/TimeOut"
+DB_URL=mongodb+srv://sh3000444:EuACQK5n92JI3vn4@cluster0.2ttcjcg.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0
```
Comment (SariGlick) [inline server-side/.env:3]: remove changes from this file

File: client-side/src/i18n.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+import languageDetector from 'i18next-browser-languagedetector';
+import Backend from 'i18next-http-backend'
+i18n
+.use(initReactI18next)
+.use(languageDetector)
+.use(Backend)
+.init(
+    {
+        debug:true,
+        fallbackLng:'he'
```
Comment (SariGlick) [inline client-side/src/i18n.js:12]: fallbackLng ahould be same as default language (that you declared as en), please change

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Review by EA171 — APPROVED** [review]

**Review by e171271 — APPROVED** [review]

**Review by SariGlick — APPROVED** [review]

**Pull request merged**